src/accelerate/utils/launch.py

```diff
@@ -46,13 +48,49 @@ def _filter_args(args, parser, default_args=[]):
     return new_args
 
 
+def _get_mpirun_args():
+    """
+    Determines the executable and argument names for mpirun, based on the type of install. The supported MPI programs
+    are: OpenMPI, Intel MPI, or MVAPICH.
+
+    Returns: Program name and arg names for hostfile, num processes, and processes per node
+    """
+    # Find the MPI program name
+    mpi_apps = [x for x in ["mpirun", "mpiexec"] if which(x)]
+
+    if not mpi_apps:
+        raise OSError("mpirun or mpiexec were not found. Ensure that Intel MPI, Open MPI, or MVAPICH are installed.")
+
+    # Call the app with the --version flag to determine which MPI app is installed
+    mpi_app = mpi_apps[0]
+    mpirun_version = subprocess.check_output([mpi_app, "--version"])
+
+    if b"Open MPI" in mpirun_version:
+        return mpi_app, "--hostfile", "-n", "--npernode"
+    else:
+        # Intel MPI and MVAPICH both use the same arg names
+        return mpi_app, "-f", "-n", "-ppn"
+
+
 def prepare_simple_launcher_cmd_env(args: argparse.Namespace) -> Tuple[List[str], Dict[str, str]]:
     """
     Prepares and returns the command list and an environment with the correct simple launcher environment variables.
     """
     cmd = []
     if args.no_python and args.module:
         raise ValueError("--module and --no_python cannot be used together")
+
+    mpirun_hostfile = getattr(args, "mpirun_hostfile", None)
+
+    if mpirun_hostfile:
```

Comment:
`args` will always have `mpirun_hostfile`, so let's just use it
```suggestion
    if args.mpirun_hostfile is not None:
```